### ISSUE TITLE
False positive no-unused-vars when importing typescript type in .vue file

### ISSUE THREAD

**Issue by WIStudent** [vuejs/eslint-config-typescript#14]
I am getting a false positive error `no-unused-vars` when importing and using a typescript type in a .vue file.
My `eslintrc.js`:
```javascript
module.exports = {
  root: true,
    env: {
      node: true
    },
    extends: [
      "plugin:vue/vue3-essential",
      "eslint:recommended",
      "@vue/typescript"
    ],
    parserOptions: {
      "parser": "@typescript-eslint/parser"
    },
    rules: {
      "@typescript-eslint/semi": ["error"]
    }
};
```

One possible workaround I found is to disable the `no-used-vars` rule for .vue files and enable `@typescript-eslint/no-unused-vars`:
```javascript
overrides: [
  // Fix no-used-vars when importing ts types in .vue files
  {
    files: ["*.vue"],
    rules: {
      'no-unused-vars': 'off',
      '@typescript-eslint/no-unused-vars': 'error'
    }
  }
]
```

**Comment by ux-engineer** [comment]
Same here and not just for imported types...

**Comment by Livijn** [comment]
I am also receiving the same error. Here is a reproducible example.

```
<script lang="ts">
import { defineComponent, PropType } from 'vue';

export default defineComponent({
  name: 'HelloWorld',
  props: {
    msg: String as PropType<String>
  }
})
</script>
```

Throws error:
`ESLint: 'PropType' is defined but never used.(no-unused-vars)`

**Comment by backbone87** [comment]
I have a similar problem with eslint core rule `no-undef` and global types.

The reason is that some eslint core rules should be disabled.
- `no-undef` is disabled by [`plugin:@typescript-eslint/eslint-recommended`](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/src/configs/eslint-recommended.ts)
- `no-unused-vars` is disabled by [`plugin:@typescript-eslint/recommended`](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/src/configs/recommended.ts) (also extends the above config)

eslint-config-typescript only extends `plugin:@typescript-eslint/eslint-recommended`, but not `plugin:@typescript-eslint/recommended`, so `no-unused-vars` is not disabled. So extending `plugin:@typescript-eslint/recommended` should fix this issue.

In my case `no-undef` gets triggered inside vue files anyway. The reason for this is that `plugin:@typescript-eslint/eslint-recommended` only disables core rules for `.ts/x` files, but not `.vue` files. Except for copying over the rule disables, i have no solution here, yet.

**Comment by pr3tori4n** [comment]
I'm seeing similar issues in our codebase. Can verify there's issues with `no-unused-vars`. Can verify that @backbone87 workaround also fixes those.

Also was seeing issues with globals. I was able to resolve those by adding them to the globals in the config
```
globals: {
  foo: 'readonly',
}
```

**Comment by klarkc** [comment]
Improving @backbone87  suggestion, to disable this rule only in vue files:

```json
{
"overrides": [
    {
      "files": ["*.vue"],
      "rules": {
        "no-unused-vars": "off"
      }
    }
  ]
}
```

**Comment by agentschmitt** [comment]
i have also problem with the rule no-undef when using the ruleset @vue/typescript/recommended

`$ vue-cli-service lint --no-fix
error: 'NotificationPermission' is not defined (no-undef) at src\views\Navigation.vue:72:28:
  70 |     }
  71 |
> 72 |     public notification(): NotificationPermission {
     |                            ^
  73 |         return Notification.permission;
  74 |     }
  75 |`
